nixos/modules/services/networking/sunshine.nix

```diff
@@ -0,0 +1,148 @@
+{ config, lib, pkgs, utils, ... }:
+let
+  inherit (lib) mkEnableOption mkPackageOption mkOption mkIf types optionals getExe;
+  inherit (utils) escapeSystemdExecArgs;
+  cfg = config.services.sunshine;
+
+  # ports used are offset from a single base port, see https://docs.lizardbyte.dev/projects/sunshine/en/latest/about/advanced_usage.html#port
+  generatePorts = port: offsets: map (offset: port + offset) offsets;
+  defaultPort = 47989;
+
+  appsFormat = pkgs.formats.json { };
+  settingsFormat = pkgs.formats.keyValue { };
+
+  appsFile = appsFormat.generate "apps.json" cfg.applications;
+  configFile = settingsFormat.generate "sunshine.conf" (cfg.settings // (if cfg.applications.apps == [ ] then { } else { file_apps = appsFile; }));
```

Comment:
Maybe we can make use of the module system here.

We could add the following block to `config` and simplify the line above by only looking at `cfg.settings`:

```nix
services.sunshine.settings.file_apps = mkIf (cfg.applications.apps != [ ]) appsFile;
```

This still throws an error if users change `services.sunshine.settings.file_apps` instead of silently ignoring the value.

Reply:
This is a neat idea. It's pretty clean. Updated, thanks.